app/src/main/java/ch/epfl/sdp/peakar/user/challenge/goal/RemotePointsChallenge.java

```diff
@@ -42,51 +55,206 @@ public static Challenge generateNewChallenge(String founderID, long goalPoints)
+
+        //Add challenge status
+        challengeRef.child(Database.CHILD_CHALLENGE_STATUS).setValue(ChallengeStatus.PENDING.getValue());
+
+        // Join the new challenge remotely + set value to current amount of points that the user has
+        Database.getInstance().getReference().child(Database.CHILD_USERS).child(founderID).child(Database.CHILD_CHALLENGES).child(id).setValue(0);
+        RemotePointsChallenge newChallenge = new RemotePointsChallenge(id, users, 0L, ChallengeStatus.PENDING.getValue(),
+                            creationTime, durationInDays,null,null);
         // Add locally if the founder is the authenticated user
         if(founderID.equals(AuthService.getInstance().getID())) AuthService.getInstance().getAuthAccount().getChallenges().add(newChallenge);
         return newChallenge;
     }
 
+
+    /**
+     * @return true if the challenge expired, false if not
+     */
+    @SuppressLint("NewApi")
+    public boolean isChallengeFinished(){
+       LocalDateTime finishDateTime = getFinishDateTime();
+        return finishDateTime.compareTo(LocalDateTime.now()) < 0;
+    }
+
+    @SuppressLint("NewApi")
     @Override
     public ChallengeOutcome join() {
         // If the authenticated user has already joined this challenge.
         if(getUsers().contains(AuthService.getInstance().getID())) return ChallengeOutcome.NOT_POSSIBLE;
+
         // Join remotely
         Database.getInstance().getReference().child(Database.CHILD_CHALLENGES).child(getID()).child(Database.CHILD_USERS).child(AuthService.getInstance().getID()).setValue(JOINED);
         // Add locally
         AuthService.getInstance().getAuthAccount().getChallenges().add(this);
+
+        /*Set start score */
+        Database.getInstance().getReference().child(Database.CHILD_USERS).child(AuthService.getInstance().getID()).child(Database.CHILD_CHALLENGES)
+                .child(getID()).setValue(AuthService.getInstance().getAuthAccount().getScore());
+
+        //Check if challenge has started
+        if(getStatus() == ChallengeStatus.PENDING.getValue()){
+            //Start challenge status and start/stop (locally)
+            setStatus(ChallengeStatus.ONGOING);
+            setStartDateTime(LocalDateTime.now());
+            setFinishDateTime(LocalDateTime.now().plusDays(getDurationInDays()));
+            //Start challenge status and start/stop (on DB)
+            Database.getInstance().getReference().child(Database.CHILD_CHALLENGES).child(getID()).child(Database.CHILD_CHALLENGE_START).setValue(getStartDateTime().toString());
+            Database.getInstance().getReference().child(Database.CHILD_CHALLENGES).child(getID()).child(Database.CHILD_CHALLENGE_FINISH).setValue(getFinishDateTime().toString());
+            Database.getInstance().getReference().child(Database.CHILD_CHALLENGES).child(getID()).child(Database.CHILD_CHALLENGE_STATUS).setValue(ChallengeStatus.ONGOING.getValue());
```

Comment:
is it worth to save a variable like 
child = Database.getInstance().getReference().child(Database.CHILD_CHALLENGES).child(getID())
To use on all three function calls?

Reply:
I'll prefere to let that as it is, It is clearer since the path to the DB child is directly visible